v2/pkg/protocols/protocols.go

```diff
@@ -61,9 +62,16 @@ type ExecuterOptions struct {
 
 	Operators []*operators.Operators // only used by offlinehttp module
 
+	Colorizer      aurora.Aurora
 	WorkflowLoader model.WorkflowLoader
 }
 
+// Copy returns a copy of the executeroptions structure
+func (e ExecuterOptions) Copy() ExecuterOptions {
+	copy := e
```

Comment:
The struct contains non-primitive fields, hence you won't be able to create a full copy this way. If retaining references between the "copies" is acceptable, then it should be documented, or even the method renamed to "ShallowCopy" or "CopyShallow". You could also describe why this method is needed.

Reply:
It's intentional to do a shallow copy since all we want to do is want to avoid modifying the ExecuterOptions structure itself across different protocols, but the objects inside it are safe for use.